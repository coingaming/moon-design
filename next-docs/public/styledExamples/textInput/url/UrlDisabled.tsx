import React from 'react';
import { TextInput } from '@heathmont/moon-core';

const Example = () => (
  <div className="flex flex-col lg:flex-row justify-around items-end w-full gap-2">
    <TextInput
      type="url"
      value="https://sportsbet.io"
      inputSize="medium"
      label="medium"
      hintText="Informative message holder"
      disabled
    />
    <TextInput
      type="url"
      value="https://sportsbet.io"
      inputSize="large"
      label="large"
      hintText="Informative message holder"
      disabled
    />
    <TextInput
      type="url"
      value="https://sportsbet.io"
      inputSize="xlarge"
      label="xlarge"
      hintText="Informative message holder"
      disabled
    />
  </div>
);

export default Example;
