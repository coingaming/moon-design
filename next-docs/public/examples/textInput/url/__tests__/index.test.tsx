/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { moonDesignLight, ThemeProvider } from '@heathmont/moon-themes';
import UrlWithLabel from '../UrlWithLabel';
import UrlWithoutLabel from '../UrlWithoutLabel';
import UrlWithHint from '../UrlWithHint';
import UrlDisabled from '../UrlDisabled';
import UrlError from '../UrlError';
import UrlBgColor from '../UrlBgColor';
import UrlReadonly from '../UrlReadonly';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={moonDesignLight}>{component}</ThemeProvider>
);

describe('TextInput: type Url', () => {
  it('renders with label', () => {
    const testRenderer = renderer.create(renderWithTheme(<UrlWithLabel />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders without label', () => {
    const testRenderer = renderer.create(renderWithTheme(<UrlWithoutLabel />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders with hint', () => {
    const testRenderer = renderer.create(renderWithTheme(<UrlWithHint />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders disabled', () => {
    const testRenderer = renderer.create(renderWithTheme(<UrlDisabled />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders error', () => {
    const testRenderer = renderer.create(renderWithTheme(<UrlError />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('supports background colour', () => {
    const testRenderer = renderer.create(renderWithTheme(<UrlBgColor />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders readOnly', () => {
    const testRenderer = renderer.create(renderWithTheme(<UrlReadonly />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
