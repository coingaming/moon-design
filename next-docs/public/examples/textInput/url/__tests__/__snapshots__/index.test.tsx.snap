// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput: type Url renders disabled 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF UYcZh"
      disabled={true}
    >
      <label
        className="sc-iKKmqK eRAvVB"
      >
        medium
      </label>
      <input
        className="sc-dIZBxk dPVOkx"
        disabled={true}
        placeholder=" "
        type="url"
        value="https://sportsbet.io"
      />
      <p
        className="sc-fOpBKL ithMSx"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF UYcZh"
      disabled={true}
    >
      <label
        className="sc-iKKmqK eRAvVB"
      >
        large
      </label>
      <input
        className="sc-dIZBxk jythDm"
        disabled={true}
        placeholder=" "
        type="url"
        value="https://sportsbet.io"
      />
      <p
        className="sc-fOpBKL ithMSx"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF UYcZh"
      disabled={true}
    >
      <div
        className="sc-MEQRl kBJTEJ"
      >
        <input
          className="sc-dIZBxk gGzdqY"
          disabled={true}
          placeholder=" "
          type="url"
          value="https://sportsbet.io"
        />
        <label
          className="sc-khrHlC jpbMtc"
        >
          xlarge
        </label>
      </div>
      <p
        className="sc-fOpBKL ithMSx"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Url renders error 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <label
        className="sc-iKKmqK eRAvVB"
      >
        medium
      </label>
      <input
        className="sc-dIZBxk emmGGf"
        placeholder="e.g. https://sportsbet.io"
        type="url"
      />
      <p
        className="sc-fOpBKL eMFfDR"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <label
        className="sc-iKKmqK eRAvVB"
      >
        large
      </label>
      <input
        className="sc-dIZBxk MqtJU"
        placeholder="e.g. https://sportsbet.io"
        type="url"
      />
      <p
        className="sc-fOpBKL eMFfDR"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <div
        className="sc-MEQRl kBJTEJ"
      >
        <input
          className="sc-dIZBxk jKqydC"
          placeholder="e.g. https://sportsbet.io"
          type="url"
        />
        <label
          className="sc-khrHlC jpbMtc"
        >
          xlarge
        </label>
      </div>
      <p
        className="sc-fOpBKL eMFfDR"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Url renders readOnly 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <label
        className="sc-iKKmqK eRAvVB"
      >
        medium
      </label>
      <input
        className="sc-dIZBxk dPVOkx"
        placeholder=" "
        readOnly={true}
        type="url"
        value="https://sportsbet.io"
      />
      <p
        className="sc-fOpBKL ithMSx"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <label
        className="sc-iKKmqK eRAvVB"
      >
        large
      </label>
      <input
        className="sc-dIZBxk jythDm"
        placeholder=" "
        readOnly={true}
        type="url"
        value="https://sportsbet.io"
      />
      <p
        className="sc-fOpBKL ithMSx"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <div
        className="sc-MEQRl kBJTEJ"
      >
        <input
          className="sc-dIZBxk gGzdqY"
          placeholder=" "
          readOnly={true}
          type="url"
          value="https://sportsbet.io"
        />
        <label
          className="sc-khrHlC jpbMtc"
        >
          xlarge
        </label>
      </div>
      <p
        className="sc-fOpBKL ithMSx"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Url renders with hint 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <label
        className="sc-iKKmqK eRAvVB"
      >
        medium
      </label>
      <input
        className="sc-dIZBxk dPVOkx"
        placeholder="e.g. https://sportsbet.io"
        type="url"
      />
      <p
        className="sc-fOpBKL ithMSx"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <label
        className="sc-iKKmqK eRAvVB"
      >
        large
      </label>
      <input
        className="sc-dIZBxk jythDm"
        placeholder="e.g. https://sportsbet.io"
        type="url"
      />
      <p
        className="sc-fOpBKL ithMSx"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <div
        className="sc-MEQRl kBJTEJ"
      >
        <input
          className="sc-dIZBxk gGzdqY"
          placeholder="e.g. https://sportsbet.io"
          type="url"
        />
        <label
          className="sc-khrHlC jpbMtc"
        >
          xlarge
        </label>
      </div>
      <p
        className="sc-fOpBKL ithMSx"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Url renders with label 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <label
        className="sc-iKKmqK eRAvVB"
      >
        medium
      </label>
      <input
        className="sc-dIZBxk dPVOkx"
        placeholder="e.g. https://sportsbet.io"
        type="url"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <label
        className="sc-iKKmqK eRAvVB"
      >
        large
      </label>
      <input
        className="sc-dIZBxk jythDm"
        placeholder="e.g. https://sportsbet.io"
        type="url"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <div
        className="sc-MEQRl kBJTEJ"
      >
        <input
          className="sc-dIZBxk gGzdqY"
          placeholder="e.g. https://sportsbet.io"
          type="url"
        />
        <label
          className="sc-khrHlC jpbMtc"
        >
          xlarge
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Url renders with label RTL 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <label
        className="sc-iKKmqK eRAvVB"
        dir="rtl"
      >
        medium
      </label>
      <input
        className="sc-dIZBxk dPVOkx"
        dir="rtl"
        placeholder="e.g. https://sportsbet.io"
        type="url"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <label
        className="sc-iKKmqK eRAvVB"
        dir="rtl"
      >
        large
      </label>
      <input
        className="sc-dIZBxk jythDm"
        dir="rtl"
        placeholder="e.g. https://sportsbet.io"
        type="url"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <div
        className="sc-MEQRl kBJTEJ"
      >
        <input
          className="sc-dIZBxk gGzdqY"
          dir="rtl"
          placeholder="e.g. https://sportsbet.io"
          type="url"
        />
        <label
          className="sc-khrHlC ckbtXb"
        >
          xlarge
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Url renders without label 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <input
        className="sc-dIZBxk dPVOkx"
        placeholder="medium"
        type="url"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <input
        className="sc-dIZBxk jythDm"
        placeholder="large"
        type="url"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <div
        className="sc-MEQRl kBJTEJ"
      >
        <input
          className="sc-dIZBxk fptWeR"
          placeholder="xlarge"
          type="url"
        />
        <label
          className="sc-khrHlC jpbMtc"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Url supports background colour 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <label
        className="sc-iKKmqK eRAvVB"
      >
        medium
      </label>
      <input
        className="sc-dIZBxk kOuPuS"
        placeholder="e.g. https://sportsbet.io"
        type="url"
      />
      <p
        className="sc-fOpBKL ithMSx"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <label
        className="sc-iKKmqK eRAvVB"
      >
        large
      </label>
      <input
        className="sc-dIZBxk iEiYBx"
        placeholder="e.g. https://sportsbet.io"
        type="url"
      />
      <p
        className="sc-fOpBKL ithMSx"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="sc-oTTbF eQlMmT"
    >
      <div
        className="sc-MEQRl cVOSay"
      >
        <input
          className="sc-dIZBxk gGzdqY"
          placeholder="e.g. https://sportsbet.io"
          type="url"
        />
        <label
          className="sc-khrHlC jpbMtc"
        >
          xlarge
        </label>
      </div>
      <p
        className="sc-fOpBKL ithMSx"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;
